src/index.js

```diff
@@ -37,6 +38,15 @@ export function init(getLangLoaderFn, configObj = {}) {
   return new Promise(loadBundle)
 }
 
+/**
+ * Allow updating the language messages/translations after init.
+ * This is beneficial in project with two language sources.
+ * @param {Function} updateLangLoaderFn A function that resolves new messages array to update the existing messages
+ */
+export function updateLangMessages(updateLangLoaderFn) {
```

Comment:
Thank you for your PR in the first place, I roughly idealised what you have achieved here:

**Goal: Initilize i18n including additionally message from outside library**

1. `i18n.setUpdateLoader` is called with the components message loading functionality
2. then the main app calls `i18n.init` with a loader functionality to load its messages
3. after main is initialised its message, i18n will conditionally load the `updateLoader` messages
4. i18n initialisation is done

That's how I read your code.

---

Question: wouldn't this be achievable without even modifying i18n? 
Please correct me if I'm wrong here, but here's my code how I imagined it.

```js
import { init } from '@signavio/i18n'

const localeLoader = (localeToLoad) => {
    return async (callback) => {
        try {
            const languageLocale = await import(`./local/locales/${localeToLoad}.js`)

            // additionally loading library's message and merging them afterwards
            const myLibraryLocales = await import(`<my-library>/locales/${localeToLoad}.js`)
            // calling i18n with a merged set of both locales, thus getting it ready for final usage
            callback({ ...languageLocale, ...myLibraryLocales })
        } catch (e) {
            // ... fallback handling
        }
    }
}

await init(localeLoader, "de_de")
```

Reply:
@marpme , thanks for the reply and the suggestion. Yes this look like a solution to our problem. I was a little concerned this will affect the way we lazy-load load the locales though in the current `pex-client` implementations. Until we test the effect of this solution in the real implementation , I will close the this PR for the time being. Thanks once again.